flow-report/src/summary/highest-impact.tsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license Copyright 2021 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {FunctionComponent} from 'preact';
+import {useMemo} from 'preact/hooks';
+
+import {Util} from '../../../report/renderer/util';
+import {useFlowResult} from '../util';
+
+const MAX_HIGH_IMPACT = 5;
+
+function computeHighestImpactAudits(reportResults: LH.ReportResult[], categoryId: string) {
+  const reportListMap: Map<string, {
+    ref: LH.ReportResult.AuditRef,
+    reports: number[],
+    /** Total extra score that would be gained if this audit passed in every report. */
+    remainingScore: number,
+  }> = new Map();
+
+  for (let i = 0; i < reportResults.length; ++i) {
+    const reportResult = reportResults[i];
+    const category = reportResult.categories[categoryId];
+    if (!category) continue;
+
+    for (const auditRef of category.auditRefs) {
+      const reportListForAudit = reportListMap.get(auditRef.id) || {
+        ref: auditRef,
+        reports: [],
+        remainingScore: 0,
+      };
+      if (!Util.showAsPassed(auditRef.result)) {
+        reportListForAudit.reports.push(i);
+        reportListForAudit.remainingScore += (1 - Number(auditRef.result.score)) * auditRef.weight;
```

Comment:
This is a slightly than https://github.com/GoogleChrome/lighthouse/pull/13082#discussion_r712381050. I figured `auditRef.weight` makes more sense as a weight than a tiebreaker. # of reports is now the tiebreaker. WDYT @patrickhulce?

Reply:
I like the concept, I think we'll also want a version of this for perf though where everything has weight 0 but we still want to consider the score.

I'm thinking...

```
interface AuditEntry {
  occurrences: Array<{ref: AuditResultRef, reportIndex: number}>
  totalCategoryScoreHeadroom: number // this one is weighted by `.weight`
  totalAuditScoreHeadroom: number // this one is not
  totalOverallSavingsMs?: number
}